client/scss/components/_dropdown-button.scss

```diff
@@ -36,7 +36,7 @@ $radius: theme('borderRadius.sm');
     --last-item-border-top: #{$separator};
   }
 
-  &:has([data-placement^='bottom']) {
+  &:has(:not([hidden]) > [data-placement^='bottom']) {
```

Comment:
This fixes an issue where we lose the top rounded corners after opening and closing the footer actions:

<img width="1006" alt="image" src="https://github.com/user-attachments/assets/7a61dbe4-408e-40cc-9b17-17a5fef0f23a">

It's probably caused by the "keep mounted" behaviour I added in bbb083a18385082046f723d94fc3191dcb439ad2.

There's another way we can fix this, which is:

```suggestion
  &:has([data-state="visible"][data-placement^='bottom']) {
```

which, while it looks nicer, had a different behaviour due to the `data-state` attribute being set immediately instead of after the transition has completed. The removal of the border caused a slight flash of height change in the last button in the dropdown.

Meanwhile, the `hidden` attribute is set after the transition finishes, and I found this to be the closest behaviour to what we had in 6.1.